src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -58,8 +59,28 @@ public class BodyModelBinder : BindingSourceModelBinder
                 return new ModelBindingResult(null, bindingContext.ModelName, isModelSet: false);
             }
 
-            var model = await formatter.ReadAsync(formatterContext);
+            object model = null;
+            try
+            {
+                model = await formatter.ReadAsync(formatterContext);
+            }
+            catch (Exception ex)
+            {
+                model = GetDefaultValueForType(bindingContext.ModelType);
```

Comment:
FYI: following a similar approach as in Web API
Example:
`ReadFromStream` method in
http://aspnetwebstack.codeplex.com/SourceControl/latest#src/System.Net.Http.Formatting/Formatting/XmlMediaTypeFormatter.cs